### PR TITLE
fixes footer and header heights on ios 11 to be automatic dimension if the view returns as 0

### PR DESCRIPTION
**Pull request by rhaining**

### CODE REVIEW

File: Static/DataSource.swift [hunk 1]
```diff
@@ -204,7 +204,14 @@ extension DataSource: UITableViewDataSource {
...
     public func tableView(_ tableView: UITableView, heightForHeaderInSection sectionIndex: Int) -> CGFloat {
-        return section(at: sectionIndex)?.header?.viewHeight ?? 0
+        var headerHeight: CGFloat = 0
+        if let headerView = section(at: sectionIndex)?.header {
+            headerHeight = headerView.viewHeight ?? UITableViewAutomaticDimension
+            if headerHeight == 0 {
+                headerHeight = UITableViewAutomaticDimension
+            }
+        }
+        return headerHeight
```
Comment (hyperspacemark) [inline Static/DataSource.swift]: Okay I think this makes sense. Can we simplify a bit using a guard?

```swift
guard let headerView = section(at: sectionIndex)?.header, let height = headerView.height, height > 0 else {
    return UITableViewAutomaticDimension
}

return height
```

**Comment by hyperspacemark** [comment]
I'm good with this. Please squash down to one commit before merging. 🎱

File: Static/DataSource.swift [hunk 2]
```diff
@@ -204,7 +204,13 @@ extension DataSource: UITableViewDataSource {
     }
 
     public func tableView(_ tableView: UITableView, heightForHeaderInSection sectionIndex: Int) -> CGFloat {
-        return section(at: sectionIndex)?.header?.viewHeight ?? 0
+        guard let headerView = section(at: sectionIndex)?.header else { return 0 }
+
+        var headerHeight = headerView.viewHeight
+        if headerHeight == 0 {
```
Comment (dmiluski) [inline Static/DataSource.swift]: After reading through the documentation, I'm not so sure we should be setting this to automatic given value of zero. Why are we setting to auto here?

    open var sectionFooterHeight: CGFloat // default is UITableViewAutomaticDimension

**Comment by txaiwieser** [comment]
I really don't think this changes are good. This 2.0.2 version changed all my layouts that were fine in iOS 11.. I think you app code its wrong, not this lib

**Comment by dmiluski** [comment]
@txaiwieser for context, how are you setting you footers? Sizing them between iOS10/11?

**Comment by txaiwieser** [comment]
I have a lot of tableViews with custom headers and footers and they were working fine in iOS 10 and 11. They have the correct sizing, but now with this new changes on the lib they are wrong even on iOS 10

(Building with Xcode 9)

**Comment by dmiluski** [comment]
@txaiwieser Are you manually laying out their sizes in layoutSubviews or something similar? Or leveraging auto layout for their sizing?

**Comment by txaiwieser** [comment]
Have you tried this using .groupped tableViews? all the sectionHeaders and sectionsFooters are now 0 height :/

Am I missing something?

**Comment by dmiluski** [comment]
@txaiwieser I'm experiencing the same thing.

I had begun an alternative approach here, which reverts to previous implementation with an addition of potential for auto layout support:
https://github.com/venmo/Static/pull/110

But ran into issues in which iOS11 was friendly to autosized sectionFooters, but iOS10 ended up clipping views, not respecting their auto layout provided intrinsic heights.

If you're on any public slack chats, would love to follow up with you there to see if we are experiencing the same thing. Otherwise, I can invite you to chat in the slack channel: iosdevelopers.slack.com

**Comment by txaiwieser** [comment]
@dmiluski Yes, I'm only on ios-developers.slack.com, but if you can sent me an invitation "txaidw@gmail.com"

**Pull request merged**